ditto/src/main/java/eu/xenit/testing/ditto/internal/DefaultNode.java

```diff
@@ -67,19 +78,36 @@ private DefaultNode(NodeBuilder builder) {
         this.txnId = builder.txnId;
         this.nodeRef = builder.nodeRef();
         this.type = builder.type;
-        this.properties = new DefaultNodeProperties(builder.context.defaultLocale, builder.properties);
+        this.properties = new DefaultNodeProperties(builder.properties);
+        this.qName = builder.qname != null ? builder.qname : Content.createQName(this.getName());
         this.aspects = new HashSet<>(builder.aspects);
+        this.childNodeCollection = new DefaultParentChildNodeCollection(this);
+
+        if (builder.context.getParent() != null && builder.context.getParentChildAssocType() != null) {
+            this.primaryParentAssoc = new DefaultParentChildAssoc(builder.context.getParent(),
+                    builder.context.getParentChildAssocType(), this, true);
+        } else {
+            this.primaryParentAssoc = null;
+        }
 
         init.accept(this, builder.context);
     }
 
-    public static NodeBuilder builder(TransactionContext context) {
-        return new NodeBuilder(context);
+    public static NodeBuilder builder(TransactionContext context, Node parent,
+            QName assocType) {
+        return new NodeBuilder(context, parent, assocType);
     }
 
     @Override
     public String getName() {
-        return (String) this.getProperties().get(Content.NAME);
+        String name = (String) this.getProperties().get(Content.NAME);
+        return name != null ? name : this.getNodeRef().getUuid();
```

Comment:
I think we should document that the uuid is returned if the name is not available

Reply:
That's standard Alfresco behavior?

Reply:
ok